test/e2e-tests.sh

```diff
@@ -33,6 +33,11 @@ failed=0
 header "Running Go e2e tests"
 go_test_e2e -timeout=20m ./test || failed=1
 
+# In case of failure output the pod logs for taskruns
+# TODO(afrittoli) This could be narrowed down to what is relevant for the failed tests
+
+(( failed )) && output_pods_logs taskrun
```

Comment:
I think we could try to do that directly in the go code (in the teardown of the tests or something)

Reply:
`tearDown` would probably be a great place to put this! https://github.com/tektoncd/pipeline/blob/ffbb67959dec0abb0f07b1d456398a99531d89f8/test/init_test.go#L67

and we already have [this confusingly named `CollectBuildLogs` function](https://github.com/tektoncd/pipeline/blob/master/test/build_logs.go#L30) (which looks to me like it's still only using init containers and therefore has a bug :S https://github.com/tektoncd/pipeline/blob/master/test/build_logs.go#L45)

Reply:
oh we didn't update this one... :sweat:

Reply:
Yeah, definitely. This has it is now has no effect as it is now, since everything has been cleaned up. I'll update the PR tomorrow.

Reply:
The helm test is already collecting pod logs, but they are not showing because it's looking for the initContainers. I think for now I will just fix that and the pod logs should be back.
Something we may want to do is remove the pod logging from specific tests and move it into the teardown so that we automatically get pod logs in case of failure  for new tests as well.

Reply:
Heh, @vdemeester do you want me to move the logging to tearDown in the PR? I can add it as a commit on top.

Reply:
Okay :angel: I am :+1: on adding a commit on top :angel:

Reply:
Done